source/admin_manual/list_of_events.rst

```diff
@@ -1349,6 +1352,73 @@ Intermediate event emitted when an HTTP request is being retried.
 
 The ``http_request_finished`` event is still sent at the end of the request.
 
+
+HTTP Server
+===========
+
+These events are emitted by Dovecot's internal HTTP library when serving
+requests (e.g. doveadm HTTP API).
+
+Common fields
+-------------
+
+Fields present in all HTTP server events.
+
++---------------------+------------------------------------------------------+
+| Field               | Description                                          |
++=====================+======================================================+
+| Inherits from :ref:`event_incomming_conn`                                  |
++---------------------+------------------------------------------------------+
+| request_id          | Assigned ID if of the received request.              |
++---------------------+------------------------------------------------------+
+| method              | HTTP verb used uppercased, e.g. ``GET``.             |
++---------------------+------------------------------------------------------+
+| target              | Request path with parameters, e.g.                   |
+|                     | ``/path/?delimiter=%2F&prefix=test%2F``.             |
++---------------------+------------------------------------------------------+
+
+http_server_request_started
+---------------------------
+
+.. versionadded:: v2.3.17
+
+Emitted when a new HTTP request is received and the request headers
+(but not body payload) are parsed.
+
+http_server_request_finished
+----------------------------
+
+.. versionadded:: v2.3.17
+
+Emitted when the HTTP request is fully completed i.e the incoming request body
+is read and its the response sent.
+
++---------------------+------------------------------------------------------+
+| bytes_in            | Amount of data read, in bytes.                       |
++---------------------+------------------------------------------------------+
+| bytes_out           | Amount of data written, in bytes.                    |
++---------------------+------------------------------------------------------+
+
+http_server_response_started
+----------------------------
+
+.. versionadded:: v2.3.17
+
+Emitted when a response to HTTP request is started.
+
+http_server_response_finished
+-----------------------------
+
+.. versionadded:: v2.3.17
+
+Emitted when the full response to the HTTP request has been sent to the client.
+
++---------------------+------------------------------------------------------+
+| bytes_in            | Amount of data read, in bytes.                       |
```

Comment:
Maybe `Amount of request data read, in bytes.`

Reply:
Added, changed bytes_out to be explicit as well.

Although, I've been wondering about bytes_in for `http_server_response_finished`. I think it will always be zero?

Reply:
I was thinking that both bytes_in and bytes_out would be identical in http_server_request_finished and http_server_response_finished.

Reply:
Although now I am wondering if there's any point at all in having http_server_response_* events?

Reply:
Originally I thought that it would work like:
 * http_server_request_started
 * http_server_request_finished
 * http_server_response_started
 * http_server_response_finished

But since it's now actually: 
 * http_server_request_started
 * http_server_response_started
 * http_server_response_finished
 * http_server_request_finished

And all of the last 3 are sent about the same time, unless the response is really long. So the response events seem rather unnecessary.